src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            BulldozerProducer bulldozer = new BulldozerProducer();
+            return new ArrayList<>(bulldozer.get());
+        }
+        if (type.equals(Excavator.class)) {
+            ExcavatorProducer excavator = new ExcavatorProducer();
+            return new ArrayList<>(excavator.get());
+        }
+        if (type.equals(Truck.class)) {
+            TruckProducer truck = new TruckProducer();
+            return new ArrayList<>(truck.get());
+        }
+        return Collections.emptyList();
```

Comment:
The `getAll(Class type)` method can be improved by creating a local variable for the machines list. This can make the code cleaner and easier to read. Here's an example: 
java
List<? extends Machine> machines = someProducer.get();
return new ArrayList<>(machines);
```